cmd/manager/main.go

```diff
@@ -87,21 +85,15 @@ func main() {
 		os.Exit(1)
 	}
 
-	ctx := context.TODO()
-	// Become the leader before proceeding
-	err = leader.Become(ctx, "hostpath-provisioner-operator-lock")
```

Comment:
I fear that something like this can be dangerous during the first upgrade:
the old operator is keeping an old style lock while the new one is using the new mechanism so I fear we can have an upgrade a rolling upgrade where the old operator and the new one are fighting to control the same resources.

Reply:
So I tested this with a cluster that has 0.9 (old style) and then upgrade to the new style (this PR) and made sure that the old pod gets terminated and the new pod eventually gets the leader.

Reply:
After think about it a bit, I don't think it will be a problem in the end.

Reply:
Reason is, the operator doesn't have permissions for the lease at this point, and thus cannot use that method to determine who is leader. So it must use the configmap method. Which is the same method used by the old operator and thus they will not both run at once.

Reply:
So I ran the experiment where I had an old 0.9 operator running, and it exited here:
```
I1122 16:19:12.803484       1 request.go:621] Throttling request took 1.040336915s, request: GET:https://10.96.0.1:443/apis/autoscaling/v1?timeout=32s
{"level":"info","ts":1637598129.4038222,"logger":"controller","msg":"Stopping workers","controller":"hostpathprovisioner-controller"}
rpc error: code = NotFound desc = could not find container "fc9a4009891be56104bd8cdc2a7fc79faafcf23ca5c97431ae64b18bff341d16": container with ID starting with fc9a4009891be56104bd8cdc2a7fc79faafcf23ca5c97431ae64b18bff341d16 not found: ID does not exist
```
and the new pod started, and it acquired leader here:
```
E1122 16:22:16.322299       1 leaderelection.go:361] Failed to update lock: configmaps "hostpath-provisioner-operator-lock" is forbidden: User "system:serviceaccount:hostpath-provisioner:hostpath-provisioner-operator" cannot update resource "configmaps" in API group "" in the namespace "hostpath-provisioner"
I1122 16:22:19.711420       1 leaderelection.go:253] successfully acquired lease hostpath-provisioner/hostpath-provisioner-operator-lock
```

As you can tell by the timestamps the new leader was not elected until AFTER the previous pod was exited. So I believe we are fine here.

Reply:
perfect, thanks!